### PR TITLE
Team Bill Of Materials UI

### PR DESCRIPTION
**Pull request by cstns**
## Description

- updated the api to include device/instance status
- centralized application bom components and reused them in both team and app bom
- styling
- altered the ff accordion to programmatically hide it's content instead of altering it's display property
- truncate versions instances and add a show more button
- add a version counter to each dependency
- toggle dependency package versions
- make the dependency items start in a closed state on the team bom

## Related Issue(s)

closes https://github.com/FlowFuse/flowfuse/issues/4823

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [x] Suitable unit/system level tests have been added and they pass 
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `FlowFuse/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `FlowFuse/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

**Comment by joepavitt** [comment]
Based on our chat this morning, I wasn't expecting this ready so soon - just checking it's actually ready for review, before I dive in?

**Comment by cstns** [comment]
Not yet! I did not start to change the ff-accordion. Will mark the pr as draft until i address it.

For visibility, I have reservations of using the existing layout as is because the ff-accordion does not programmatically hide it's contents which means they would get rendered in the dom each time even though not visible. This would have negative impact on big payloads and could cause the page to freeze

**Comment by joepavitt** [comment]
@cstns ready for review?

**Comment by cstns** [comment]
Yes!

**Comment by joepavitt** [comment]
Screenshot 2024-12-13 at 13 53 59

Added sorting for the version numbers within the first accordian

File: frontend/src/components/Accordion.vue [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <template>
-    <div class="ff-accordion" :class="{'open': isOpen}" data-el="accordion">
+    <div class="ff-accordion" :class="{'open': true}" data-el="accordion">
```
Comment (joepavitt) [inline frontend/src/components/Accordion.vue:2]: why hardcode this to `true`?

File: frontend/src/mixins/BOM.js
Comment (joepavitt) [inline frontend/src/mixins/BOM.js:1]: what happened to us stopping the introduction of new `mixin` files?

**Review by joepavitt — APPROVED** [review]

**Pull request merged**